libs/core/thread_support/include/hpx/thread_support/spinlock.hpp

```diff
@@ -0,0 +1,49 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2014 Peter Dimov
+//  Copyright (c) 2020 Agustin Berge
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <hpx/config.hpp>
+
+#include <atomic>
+
+namespace hpx { namespace util { namespace detail {
+
+    /// Lockable spinlock class
+    struct spinlock
+    {
+    public:
+        HPX_NON_COPYABLE(spinlock);
+
+    private:
+        std::atomic_flag m = ATOMIC_FLAG_INIT;
+
+        HPX_CORE_EXPORT void yield_k(unsigned) noexcept;
+
+    public:
+        spinlock() = default;
+
+        bool try_lock() noexcept
+        {
+            return !m.test_and_set(std::memory_order_acquire);
+        }
+
+        void lock() noexcept
+        {
+            for (unsigned k = 0; !try_lock(); ++k)
+                yield_k(k);
```

Comment:
To reduce the pressure on the atomic flag this could be rewritten as:
```
while (!try_lock())
{
    util::yield_while([this] { return is_locked(); }, 
        "hpx::util::detail::spinlock::lock");
}
```

Reply:
I'm only looking at cutting the dependency on boost's internal implementation details.

If you wish to make those changes feel free to push them to this branch.

Reply:
I changed you PR slightly to use `std::atomic<bool>` and applied the modification mentioned above.